File: src/main/scala/com/refactorings/ruby/FeatureFlag.java
package com.refactorings.ruby;

public enum FeatureFlag {
  ExampleFeature;

  private boolean currentValue = false;

  public void activateIn(Runnable closure) {
    boolean oldValue = currentValue;
    currentValue = true;
    closure.run();
    currentValue = oldValue;
  }

  public boolean isActive() {
    return currentValue;
  }
}

File: src/main/scala/com/refactorings/ruby/FeatureFlag.scala
package com.refactorings.ruby

object FeatureFlag extends Enumeration {
  val MergeSingletonClasses: FeatureFlag = FeatureFlag("merge singleton classes")

  case class FeatureFlag(name: String) extends Val(nextId, name) {
    private var currentValue = false

    def activateIn[T](closure: => T): T = {
      val oldValue = currentValue
      currentValue = true
      val result = closure
      currentValue = oldValue
      result
    }

    def isActive: Boolean = currentValue
  }
}
